instrumentation/github.com/gin-gonic/gin/otelgin/config.go

```diff
@@ -27,15 +29,35 @@ type config struct {
 	GinMetricAttributeFn GinMetricAttributeFn
 }
 
+// defaultSpanNameFormatter is the default span name formatter.
+var defaultSpanNameFormatter SpanNameFormatter = func(c *gin.Context) string {
+	method := strings.ToUpper(c.Request.Method)
+	if !slices.Contains([]string{
+		http.MethodGet, http.MethodHead,
+		http.MethodPost, http.MethodPut,
+		http.MethodPatch, http.MethodDelete,
+		http.MethodConnect, http.MethodOptions,
+		http.MethodTrace,
+	}, method) {
+		method = "HTTP"
+	}
+
+	if path := c.FullPath(); path != "" {
+		return method + " " + path
+	}
+
+	return method
+}
+
 // Filter is a predicate used to determine whether a given http.request should
 // be traced. A Filter must return true if the request should be traced.
 type Filter func(*http.Request) bool
 
 // GinFilter filters an [net/http.Request] based on content of a [gin.Context].
 type GinFilter func(*gin.Context) bool
 
-// SpanNameFormatter is used to set span name by http.request.
-type SpanNameFormatter func(r *http.Request) string
+// SpanNameFormatter is a function that takes a gin.Context and returns a string.
```

Comment:
This comment is like showing a box of chocolate and saying "this is chocolate". That's not really useful.
The previous comment, explaining why we need that type is a better comment IMHO.

```suggestion
// SpanNameFormatter is used by `WithSpanNameFormatter` to customize the request's span name.
```